src/contrib/cluster/Akka.Cluster.Sharding.Tests/ConstantRateEntityRecoveryStrategySpec.cs

```diff
@@ -0,0 +1,42 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="PersistentShard.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Immutable;
+using Akka.TestKit;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Cluster.Sharding.Tests
+{
+    using EntityId = String;
+
+    public class ConstantRateEntityRecoveryStrategySpec : AkkaSpec
+    {
+        private readonly EntityRecoveryStrategy strategy;
+
+        public ConstantRateEntityRecoveryStrategySpec()
+        {
+            strategy = EntityRecoveryStrategy.ConstantStrategy(Sys, TimeSpan.FromSeconds(1), 2);
+        }
+
+        [Fact]
+        public void ConstantRateEntityRecoveryStrategySpec_must_recover_entities()
+        {
+            var entities = ImmutableHashSet.Create<EntityId>("1", "2", "3", "4", "5");
+
+            // TODO: https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/test/scala/akka/cluster/sharding/ConstantRateEntityRecoveryStrategySpec.scala
```

Comment:
I cannot compile it rigth now, but I think, this scala code:

```scala
val startTime = System.nanoTime()
      val resultWithTimes = strategy.recoverEntities(entities).map(
        _.map(entityIds ⇒ (entityIds → (System.nanoTime() - startTime).nanos)))

      val result = Await.result(Future.sequence(resultWithTimes), 6.seconds)
        .toVector.sortBy { case (_, duration) ⇒ duration }
      result.size should ===(3)

      val scheduledEntities = result.map(_._1)
```

should be equivalent to this:

```csharp
var entities = ImmutableHashSet.Create<EntityId>("1", "2", "3", "4", "5");
var startTime = DateTime.UtcNow;
var resultWithTimes = strategy.RecoverEntities(entities)
    .Select(async bucketTask => new KeyValuePair<,>((await bucketTask), DateTime.UtcNow - startTime))
    .ToArray();

var result = Task.WaitAll(resultWithTimes).OrderBy(pair => pair.Value).ToArray();
result.Length.Should().Be(3);

var scheduledEntities = result.Select(pair => pair.Key).ToArray();
```